### ISSUE TITLE
Waiting list voucher not send out without apparent reason

### ISSUE THREAD

**Issue by fefrei** [pretix/pretix#3628]
### Problem and impact

I've managed to get into a waiting list state that is either broken, or probably just confusing. Here's the situation:

- I'm selling (free) tickets for two iterations of an event. They are created as two variations of the same product.
- I'm also offering a flexible ticket where the customer allows me to assign them a date.
- There's also internal variants of the standard tickets, bookable only with a voucher.

This is actually surprisingly easy thanks to Pretix's flexible quota system. There are three quotas: One for total available spots, and two each for the two iterations. (They're 30 + 30 = 60 spots, so the quotas sum up.) The standard tickets count into both the general and iteration-specific quota, the flexible tickets only into the general quota.

I then found myself in the following situation, which I find contradictory:

- One iteration of the event has sold out.
- This is because the event-specific quota is full:
    <img width="400" alt="image" src="https://github.com/pretix/pretix/assets/3078885/e06ee5e1-d9d6-4726-847b-a6932cb5e9ee">
- There is one user on the waiting list, waiting for a spot:
    <img width="606" alt="image" src="https://github.com/pretix/pretix/assets/3078885/3d52f9bd-9c4a-4502-bd29-78bf6aac0cc4">
- No vouchers have been automatically created. There are only manually created ones, for the internal variants of the tickets:
    ![image](https://github.com/pretix/pretix/assets/3078885/42ac55aa-108e-4a44-9b00-09423bc97976)

Note the quota page claims there are 5 vouchers reserved from the quota, but only 4 vouchers remain, and the waiting list entry does not have a voucher. These screenshots are consistent – I refreshed all pages again, nothing changed while taking the screenshots.
Before I was able to take the screenshots, the one voucher wasn't used up yet. At that point, the “Vouchers and waiting list reservations” entry showed a count of 6. I never created 6 vouchers for the product, so this clearly counted the waiting list entry.

In this state, Pretix refused to create a voucher for the waiting list entry as the product was unavailable (even when I manually clicked the button). It also showed as sold out in the online shop. I do not understand why – it looked as if the waiting list entry was somehow counted both into the “Waiting list (pending)” and “Vouchers and waiting list reservations” counts and the user blocked themselves from buying the product (which was unavailable because they themselves were on the waiting list‽

The situation eventually resolved itself. Now, _another_ user was able to purchase a ticket (without using a voucher). The quota now matches what I expect:

<img width="394" alt="image" src="https://github.com/pretix/pretix/assets/3078885/0665039f-7ce9-4615-a9b0-e3a04e12d1d6">

But of course, the user shouldn't have been able to skip the waiting list entry – provided they didn't have it in their cart already (but the quota didn't say that).

I have multiple possible explanations for what is going on:

- The ticket was blocked by being in a user's cart, but this was not showing up on the quota page. Are the numbers on the quota page eventually-consistent estimates? (If so, maybe this should be communicated on the page?)
- I just seriously misunderstood what's going on. (Maybe the labels on the quota page could be made more clear then?)
- There's a bug in waiting list handling for products with multiple quotas.

Feel free to close this if it makes no sense to try to track this down – I have no expectation of support, just trying to help :-)

### Version

2023.8.0

**Comment by fefrei** [comment]
Thanks for your response, it's appreciated!

Indeed, it looks like a voucher was redeemed properly (properly replacing an item in cart that was never booked):

<img width="193" alt="image" src="https://github.com/pretix/pretix/assets/3078885/8c692d05-9dcd-40e2-8da5-932ee4f29ac7">

That makes it all plausible, thank you! So the issue was me not looking at the filters in the waiting list view.

I thought a moment about ways to make this more obvious. I don't have a super clear answer (I think the UX is actually quite good here, I was just a bit blind), but one possible suggestion would be make the default filter in the waiting list view wider – one option would be to filter for "active" entries, meaning entries that are either waiting for a ticket or have a voucher assigned that's waiting for redemption. (The hope would be that that filter captures all entries needed to understand the state the system is in, while excluding entries that have been "settled" in either way.)

---

As an aside, I think the waiting lists are _not_ working for my double-quota-system as I'd expect: I have three waiting list entries for one instance of the event now. These entries are part of the _Waiting list (pending)_ block of the total quota, meaning that waiting list entries for the first instance of the event are actually blocking sales of tickets for the second instance.

Fixing this would probably require revisiting the design decision that waiting list entries reserve a spot in _all_ quotas, not just the ones that are full. I assume this is too complicated – but if you're open to considering that and want me to split this out into a dedicated issue, let me know!
